lms/services/exceptions.py

```diff
@@ -5,10 +5,28 @@ class ServiceError(Exception):
     """Base class for all :mod:`lms.services` exceptions."""
 
 
-class ConsumerKeyError(ServiceError):
+class LTILaunchVerificationError(ServiceError):
+    """
+    Raised when LTI launch request verification fails.
+
+    This is the base class for all LTI launch request verification errors.
+    Different subclasses of this exception class are raised for specific
+    failure types.
+    """
+
+
+class NoConsumerKey(LTILaunchVerificationError):
+    """Raised when a launch request has no ``oauth_consumer_key`` parameter."""
+
+
+class ConsumerKeyError(LTILaunchVerificationError):
```

Comment:
Can there be any problems with the consumer key other than it being missing? If not, perhaps `UnknownConsumerKeyError` might be a more obvious name, if a bit more verbose.

Reply:
`ConsumerKeyError` is meant to match Python's `KeyError` (which is raised when a given key is missing from a dictionary)